### ISSUE TITLE
Derive TryInto for &T

### ISSUE THREAD

**Issue by rinon** [JelteF/derive_more#69]
I have a situation where I need to convert a reference to an enum T into a reference to the contained variant type. I can't seem to figure out a way to do this without `impl TryFrom(&'a T) for &'a V`, which is not currently derived. What would you think of adding those impls? I think they would be pretty straightforward.

**Comment by JelteF** [comment]
T would need to implement clone on that case right?

**Comment by rinon** [comment]
I don't think so. Returning a reference to the inner element with the same lifetime as the argument shouldn't clone the element. For example (`$Repr` is the inner type, `$Thing` is the enum variant name):
```rust
impl<'a> TryFrom<&'a Value> for &'a $Repr {
    type Error = String;
    fn try_from(value: &'a Value) -> Result<Self, Self::Error> {
        match value {
            Value::$Thing(ref x) => Ok(x),
            _ => Err(format!("Only &Value::{} can be converted to &{}", stringify!($Thing), stringify!($Repr))),
        }
    }
}
```

**Comment by JelteF** [comment]
FYI this is implemented on master. Code like this will generate the TryInto for T, &T and &mut T
```rust
#[derive(Clone, Copy, TryInto)]
#[try_into(owned, ref, ref_mut)]
enum MixedInts {
    SmallInt(i32),
    NamedBigInt {
        int: i64,
    },
    TwoSmallInts(i32, i32),
    NamedBigInts {
        x: i64,
        y: i64,
    },
    Unsigned(u32),
    NamedUnsigned {
        x: u32,
    },
    Unit,
    #[try_into(ignore)]
    Unit2,
}
```

**Comment by JelteF** [comment]
This is now released as version 0.99.0. Announcement link: https://reddit.com/r/rust/comments/duptc0/derive_more_0990_released_biggest_release_ever
